src/main/java/me/botsko/prism/listeners/PrismInventoryEvents.java

```diff
@@ -152,42 +149,63 @@ public void onInventoryClick(final InventoryClickEvent event) {
         if (slotItem == null) {
             return;
         }
+        Prism.debug("HELD:" + ((heldItem != null) ? heldItem.toString() : "NULL"));
+        Prism.debug("SLOT:" +  slotItem.toString());
 
         switch (event.getClick()) {
             // IGNORE BOTTOM
             case LEFT:
                 if (isTopInv) {
-                    if (heldItem.getType() == Material.AIR) { //todo if the item under the cursor is null what to do?
+                    if (heldItem == null || heldItem.getType() == Material.AIR) {
                         if (slotItem.getType() != Material.AIR) {
                             RecordingQueue.addToQueue(ActionFactory.createItemStack(REMOVE, slotItem,
                                     slotItem.getAmount(), slot, null, containerLoc, player));
+                            Prism.debug("ACTION: " + event.getAction().name());
                         }
                     } else {
-                        if (slotItem.getType() == Material.AIR || slotItem.equals(heldItem)) {
-                            int amount = Math.min(slotItem.getType().getMaxStackSize(),
-                                    slotItem.getAmount() + heldItem.getAmount()) - slotItem.getAmount();
+                        int amount = 0;
+                        int maxStack = heldItem.getMaxStackSize();
+                        if (slotItem.getType() == Material.AIR && heldItem.getAmount() <= maxStack) {
+                                amount = heldItem.getAmount();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['if' child has incorrect indentation level 32, expected level should be 28.](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5535948)